src/lightmap.cpp

```diff
@@ -142,15 +144,17 @@ bool map::build_transparency_cache( const int zlev )
                     value *= sight_penalty;
                 }
                 float value_wo_fields = value;
-                for( const auto &fld : cur_submap->get_field( sp ) ) {
-                    const field_intensity_level &i_level = fld.second.get_intensity_level();
-                    if( i_level.transparent ) {
-                        continue;
+                Character &player_character = get_player_character();
+                if( !player_character.vision_mode_cache[IR_VISION] ) {
```

Comment:
Oops, I thought I'd reverted these changes. I'll do that in a bit.

Something is weird with the lightmap and I'm still trying to understand it. As far as I can tell, there's a ton of code about transparency that winds up doing nothing. I was looking for a way to make windows reduce transparency so that less than 100% light could get through, but what I discovered is that the whole thing gets completely flattened by a bool. The game really finely tracks the amount of light thats supposed to be passing through each tile, and then when its done it just says "if it's zero then false, if its any other number then true."

This prevents us from having proper fog and seems to be whats stopping IR from seeing through smoke. The changes here are mostly just me trying to see if there's a way around that, but I don't think there is.